pkg/roachprod/vm/ibm/README.md

```diff
@@ -0,0 +1,102 @@
+# IBM Cloud
+
+## Implementation
+
+The `roachprod` implementation for IBM Cloud is based out of the IBM official [Golang SDK](https://github.com/IBM/platform-services-go-sdk).
+
+### Account configuration
+
+A few per-region / per-az resources need to be created to allow `roachprod`
+to work properly:
+- a single resource group for all roachprod resources
+- one VPC per supported region
+- one inbound firewall rules per VPC
+- one subnets per AZ per VPC
+- one public gateway per AZ per VPC for the instances internet access
+- one single transit gateway with connections in each VPC
+
+These resources are fetched/created on demand when the provider is used.
+To speed up the provider's initialization on actions requiring only listing,
+only the resource group will be fetched. During cluster creation, all resources
+will be fetched and created.
+
+This process requires the user setting up the account the first time to have
+all required permissions to create the resources mentioned above.
+
+### Floating IP addresses
+
+Each floating IP address created is billed the monthly cost upfront.
+To avoid being billed one IP address for each machine that we spin up, we simply
+unbind the floating IP addresses when destroying the clusters and do not delete
+the unbound  IP addresses.
+
+This behavior allows us to have a pool of unbound IP addresses that we try and
+reuse them during the subsequent clusters creations.
+
+### Unsupported features
+
+Unimplemented features supported on IBM Cloud:
+- additional volumes management on a live cluster
+- snapshots
+- load balancer
+- grow/shrink clusters
+
+Unsupported features on IBM Cloud:
+- local SSDs
+- transient instances (a.k.a. `spot` instances)
+
+## Interacting with IBM Cloud
+
+While `roachprod` uses the Golang SDK, installing the CLI will be useful to
+authenticate with IBM Cloud.
+
+### Install the CLI and authenticate
+
+Download the installer from the IBM Cloud CLI [Github repository](https://github.com/IBM-Cloud/ibm-cloud-cli-release/releases/).
+
+Then, run the following command to authenticate:
+
+```bash
+ibmcloud login
```

Comment:
This only works with the passcode, right? Maybe mention that it can be generated via the Console's `Log in to CLI and API` in the upper-right corner.